app/models/tickets_edit_person.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class TicketsEditPerson < ApplicationRecord
+  self.table_name = "tickets_edit_person"
+
+  enum :status, {
+    open: "open",
+    closed: "closed",
+  }
+
+  has_one :ticket, as: :metadata
+
+  def action_user_groups(action)
+    case action
+    when :update_status
+      [UserGroup.teams_committees_group_wrt.id]
+    end
+  end
+
+  def self.create_ticket(wca_id, changes_requested, requester)
+    ActiveRecord::Base.transaction do
+      ticket_metadata = TicketsEditPerson.create!(
+        status: TicketsEditPerson.statuses[:open],
+        wca_id: wca_id,
+      )
+
+      changes_requested.each do |change|
+        case change[:field]
+        when :name
+          ticket_metadata.update!(
+            previous_name: change[:from],
+            new_name: change[:to],
+          )
+        when :country_iso2
+          ticket_metadata.update!(
+            previous_country_iso2: change[:from],
+            new_country_iso2: change[:to],
+          )
+        when :gender
+          ticket_metadata.update!(
+            previous_gender: change[:from],
+            new_gender: change[:to],
+          )
+        when :dob
+          ticket_metadata.update!(
+            previous_dob: change[:from],
+            new_dob: change[:to],
+          )
+        end
+      end
+
+      ticket = Ticket.create!(
+        name: "Edit Profile request by #{wca_id}",
+        ticket_type: Ticket.ticket_types[:edit_person],
+        metadata: ticket_metadata,
+      )
+
+      TicketStakeholder.create!(
+        ticket_id: ticket.id,
+        stakeholder_id: UserGroup.teams_committees_group_wrt.id,
+        stakeholder_type: TicketStakeholder.stakeholder_types[:user_group],
+        connection: TicketStakeholder.connections[:assigned],
+        is_active: true,
+      )
+      TicketStakeholder.create!(
+        ticket_id: ticket.id,
+        stakeholder_id: requester.id,
+        stakeholder_type: TicketStakeholder.stakeholder_types[:user],
+        connection: TicketStakeholder.connections[:cc],
+        is_active: true,
+      )
+
+      TicketLog.create!(
+        ticket_id: ticket.id,
+        log: "Ticket created.",
```

Comment:
Again, not super excited about localized strings in the DB. Since this is "only" logs, I'd be leaning to accept it, but can you maybe spend some 10 or so minutes to think about an `enum` set of log changes?

In my mind, typical log actions include "created", "moved", "reassigned" and "closed", all of which can easily be enumized.

Reply:
Hm... I can definintely make an exception about localized strings in the DB when it comes to `ticket_comments` later. There, users can write in whatever English they want and we store it.

In this case, for `ticket_logs`, I want to explore other options first. What kind of log entries do you expect in the future? I think having `ticket_logs` as a "clean" base table which only tracks standardized changes is desirable, because it helps getting an overview.

Reply:
If you really want to include lots of custom statuses, then I would recommend splitting it into two columns:
One enumized column that tracks basic, common, unified lifecycle changes (stuff like "created", "reassigned", "closed", etc. that all tickets everywhere share) and you simply add _one_ enum entry there that says `custom_change`.

Then, you introduce a second column called "details" or something in the `ticket_logs` table which will be null by default and which won't be needed for plain "opened" or "closed" logs, but where users can optionally enter more details if they want.

Reply:
But then again, we have to make sure that they don't "abuse" the custom log state as a separate, second "comment" field. I.e. it would need to be totally clear what goes into `ticket_logs` and what goes into `ticket_comments`. Is this even clearly defined at this point in time?

Reply:
(sorry for the spam) Actually, we could improve the flow even more: Instead of a column that contains "additional details" in the logs, you could add a nullable column that _optionally_ holds a foreign key that points to a `ticket_comment` if some user wants to have a "custom explanation" for a status change. That way, we can nicely link `ticket_logs` and `ticket_comments`.

And I can still get my standardized (enumized) lifecycle changes in `ticket_logs` :innocent:

Reply:
I agree with Gregor, we should avoid localized strings in DB whenever possible

Reply:
It seems like you want to log changes to tickets quite thoroughly. That's good and I support your basic intention, and I can tell you that simply logging (primary key id, ticket_id, change_string) won't be enough then. Believe me, it would become a nightmare to properly track and parse and organize these changes later on.

Please take a step back to the drawing board and think about the structure and information of `ticket_logs` on a DB design level again. What information do you want to track, and what is the most structured way to track this information?

From your existing description, I think
- `action_type`
- `acting_user` (foreign key ID)
- `acting_stakeholder` (foreign key ID)
    - There should probably be both -- one reference to an actual user that can be tracked, and one reference to a stakeholder on behalf of which the user was acting
    - This is the same line of thought as `ticket_comments`. Both have two fields (user_id, stakeholder_id). Makes four references in total and is a bit convoluted to track. Can you make a suggestion on how to "unify" these two independent fields? (i.e., store them only in one place, and reference it in the other place)

are all necessary. About the details: You should probably track details as well, for example by using a `changed_field` and `changed_value` logic. These can be custom, but still "somewhat" easy to parse.

Reply:
> Re how to avoid two references being referenced in two tables
How about having one more table named `ticket_users` with columns `user_id` and `stakeholder_id`? And then we can reference that table to both `comments` and `logs`? Basically we can have a method like following in `ticket_user.rb`:

```
def self.get(user_id, stakeholder_id)
  ticket_user = TicketUser.find_by(user_id: user_id, stakeholder_id: stakeholder_id)
  ticket_user ? ticket_user : TicketUser.new(user_id: user_id, stakeholder_id: stakeholder_id)
end
```

Will this be good to go?

I personally find this bit complicated though, but can't think of a better case.

Also, just re-thinking this again - do we really need `stakeholder_id` in the comments and logs? Is it fine if we just have `user_id`? In the UI, while mentioning the user name, we can add the user's relevant roles as well (for both logs and comments) so that whoever wants to know the stakeholder role connected with the user can get to know from there itself.

What do you think?

Reply:
Hm, this boils down to "organization-level" decisions pretty quickly. The question that we have to consider completely irrespective of the code or programming is "Do we want to keep track of the concrete stakeholder on behalf of which the user acted?"

I can think of scenarios where a WRT member files a request to have their own personal details edited (you should know that can happen :wink:), in which case this person would become a stakeholder as a simple competitor, but also be a stakeholder as a WRT member potentially (as far as the code is concerned) eligible to handle the ticket. Or a WCAT member who also happens to be a Delegate might be responsible for a competition and then submit it to WCAT, at which point they would become responsible for their own ticket.

How do we want to deal with this kind of situation on an organizational level?
1. We could just flat-out prevent it. Like build some exceptions into the stakeholder query that groups which the user is a member of should not be considered if the user is also the one who filed the ticket.
2. (my preferred option) always mark actions on tickets with the stakeholder attached. Like "this user acted on behalf of this stakeholder".

Regarding (2), it is totally up for debate and I am really open to change my mind and be convinced whether we want to keep track of this information or not. But _if_ we decide that we want it, then there's no way that we can prevent the double `(user_id, stakeholder_id)` bookkeeping.

Reply:
Yes I agree there will be cases where a user will be part of multiple stakeholder_types. But my point was that intuitively we can know on whose behalf the action was done through the `action_type`. But when I think again, it may not be practical when tickets grow so much. Hence, having both `user_id` and `stakeholder_id` in `ticket_logs` makes sense to me.

I'll go with option (2) as you suggested.

I understood your comment regarding `comments` and theoretically I think it's a good idea. I'm not 100% convinced practically because I somehow feel making `ticket_logs` as a more important thing than `ticket_comments` may not be so good, but I might be wrong. I'll think of it again once I start working on comments in a later PR.

I'll go ahead with implementation of `ticket_logs` table with columns `action_type`, `acting_user_id`, `acting_stakeholder_id`. Hope this is fine, thank you.